components/ui/Heading.tsx

```diff
@@ -0,0 +1,329 @@
+import { PropsWithChildren } from "react";
+
+type fontSize =
+  | 12
+  | 14
+  | 16
+  | 18
+  | 20
+  | 24
+  | 28
+  | 30
+  | 36
+  | 48
+  | 60
+  | 72
+  | 96
+  | 128;
+
+interface HeadingProps {
+  as: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+  size?: fontSize;
+  xxl?: fontSize;
+  xl?: fontSize;
+  lg?: fontSize;
+  md?: fontSize;
+  sm?: fontSize;
+  className?: string;
+}
+
+const Heading = ({
+  as: heading,
+  size,
+  xxl,
+  xl,
+  lg,
+  md,
+  sm,
+  className = "",
+  children,
+}: PropsWithChildren<HeadingProps>) => {
+  const getSmallSize = (size: fontSize) => {
```

Comment:
`getSmallSize`, `getMediumSize` ... functions could be moved out of function body since they don't use any variable or prop from the component. They are also almost the same with difference in class name prefixes `sm:`, `md:` etc. so it can be one function with arguments `(size: fontSize, prefix?: string)`